### PR TITLE
Update Azure SDK network API version

### LINKED ISSUE #966: Investigate impact of 2020-08-01 Network API version
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

With Network API version 2020-08-01, zone behavior for Standard SKU resources (Azure Load Balancer and Public IP addresses) will be updated such that:

when no zone is specified, a non-zonal resource is created
when a single zone is specified, a zonal resource is created 
when multiple zones are specified in a region with Availability Zones, a zone-redundant resource is created
A zone-redundant resource can only be created in regions where Availability Zones are supported. 

Previously, when no zone was specified, a zone-redundant resource was created. No changes will be made to existing Public IP or Standard Load Balancer resources with previous API versions.

https://azure.microsoft.com/en-us/updates/zone-behavior-change/

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: Updates Azure SDK network API version

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #966

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
@chewong is there any easy way to run the azure file / azure disk tests on this changeset? I don't think it's possible to trigger it directly from the PR

**Comment by chewong** [comment]
> @chewong is there any easy way to run the azure file / azure disk tests on this changeset? I don't think it's possible to trigger it directly from the PR

I don't think we have storage-related jobs set up against this repo so we can either test it locally or use a dummy PR like https://github.com/kubernetes/kubernetes/pull/101434 to test after this PR is merged.

**Comment by nader-ziada** [comment]
Are there any other tests jobs we need to run here? otherwise lgtm

**Comment by CecileRobertMichon** [comment]
I want to try testing this locally but haven't had time yet

we can also merge and use a pr test like Ernest mentioned above, that might be faster

File: config/default/manager_image_patch.yaml [hunk 1]
```diff
@@ -8,5 +8,5 @@ spec:
     spec:
       containers:
       # Change the value of image field below to your controller image URL
-      - image: gcr.io/k8s-staging-cluster-api-azure/cluster-api-azure-controller:latest
+      - image: docker.io/cerobert/cluster-api-azure-controller-amd64:dev
```
Comment (CecileRobertMichon) [inline config/default/manager_image_patch.yaml:11]: /hold

need to revert this

Comment (CecileRobertMichon) [reply config/default/manager_image_patch.yaml]: done

/hold cancel

**Comment by CecileRobertMichon** [comment]
Fixed the rebase, this is ready to go

Ran some testing locally, everything looks good

**Pull request merged**